### ISSUE TITLE
Recent update to 24.5.3 broke web server

### ISSUE THREAD

**Issue by X-Ryl669** [esphome/issues#5850]
### The problem

I had LibreTiny devices in 24.4.0 version that were working fine, the web server used to display:
![image](https://github.com/esphome/issues/assets/3277165/d59f875f-6828-4066-aec8-dbdbc11761c2)

I've updated ESPHome to 24.5.3 today, that went ok, but now, I don't see any item in the web server anymore:
![image](https://github.com/esphome/issues/assets/3277165/c7ae3efd-5612-48eb-9ea0-1993b708a9da)

The devices still work with HA through, and I can still OTA them. Only the web server is not working.

### Which version of ESPHome has the issue?

2024.5.3

### What type of installation are you using?

Home Assistant Add-on

### Which version of Home Assistant has the issue?

2024.5.5

### What platform are you using?

BK72XX

### Board

IR_remote_I_don_t_know

### Component causing the issue

web_server

### Example YAML snippet

```yaml
# All substitution not shown here, they work perfectly

esphome:
  name: irremotenorth
  friendly_name: ${device_name}

bk72xx:
  board: cb3s
  framework:
    version: dev

# Enable logging
logger:

# Enable Home Assistant API
web_server:
  port: 80
  auth:
    username: admin
    password: ${web_password}

# Enable Home Assistant API
api:
  encryption:
    key: ${api_password}

ota:
  password: ${ota_password}

wifi:
  ssid: ${wifi_ssid}
  password: ${wifi_password}
  manual_ip:
   static_ip: ${ip}
   gateway: ${gw}
   subnet: ${subnet}
   dns1: ${dns}
  # domain: ${domain}
  ##### Enable fallback hotspot (captive portal) in case wifi connection fails
  ap:
    ssid: "${device_name}"
    password: ${wifi_password}

captive_portal:


remote_transmitter:
  pin: P26
  carrier_duty_percent: 50%

sensor:
  - platform: homeassistant
    name: "Chambre Nord Temperature"
    entity_id: sensor.nspanel1_temperature
    id: nspanel1_temperature
    internal: true


climate:
  - platform: mitsubishi       
    name: "North"
    set_fan_mode: "quiet_4levels"
    supports_cool: true
    supports_heat: true
    supports_dry: true
    supports_fan_only: true
    horizontal_default: "middle"
    vertical_default: "down"    

switch:
  - platform: gpio
    name: "Blue Led"
    pin:
      number: P8
      inverted: false
  - platform: factory_reset
    name: Restart with Factory Default Settings
```

### Anything in the logs that might be useful for us?

```txt
No error message in the log
```

### Additional information

Got this in the Firefox's console:
```
Uncaught (in promise) Error: invalid template strings array
    Ct www.js:formatted:473
    Mt www.js:formatted:501
    H www.js:formatted:518
    _$AC www.js:formatted:737
    g www.js:formatted:720
    _$AI www.js:formatted:699
    qt www.js:formatted:907
    update www.js:formatted:934
    performUpdate www.js:formatted:361
    scheduleUpdate www.js:formatted:338
    _$Ej www.js:formatted:332
```
and this in Chrome console:
```
www.js:3 Uncaught (in promise) Error: invalid template strings array
    at Ct (www.js:3:335)
    at Mt (www.js:3:1057)
    at new H (www.js:3:1221)
    at T._$AC (www.js:3:4731)
    at T.g (www.js:3:4433)
    at T._$AI (www.js:3:4089)
    at qt (www.js:3:7409)
    at yt.update (www.js:3:7811)
    at yt.performUpdate (www.js:1:6064)
    at yt.scheduleUpdate (www.js:1:5716)
```

Also got the same issue with ESP8266 devices.

**Comment by pvossel** [comment]
Exact same issue for me.

**Comment by ssieb** [comment]
Try the fix for https://github.com/esphome/issues/issues/5793

**Comment by pvossel** [comment]
I can confirm. Your [workarround ](https://github.com/esphome/issues/issues/5793#issuecomment-2131383089) works @ssieb.
Guess this issue should be merged with #5793.

**Comment by ssieb** [comment]
Duplicate of https://github.com/esphome/issues/issues/5793
